crates/torii/grpc/src/server/mod.rs

```diff
@@ -789,6 +814,74 @@ impl DojoWorld {
         })
     }
 
+    async fn retrieve_tokens(
+        &self,
+        contract_addresses: Vec<Felt>,
+    ) -> Result<RetrieveTokensResponse, Status> {
+        let query = if contract_addresses.is_empty() {
+            "SELECT * FROM tokens".to_string()
+        } else {
+            format!(
+                "SELECT * FROM tokens WHERE contract_address IN ({})",
+                contract_addresses
+                    .iter()
+                    .map(|address| format!("{:#x}", address))
```

Comment:
Are we sure it's always using the same formatting without the padding? Since we store as string, just to be sure we're always comparing something unpadded is expected. 👍

Reply:
The formatted contract address is unpadded when serialized to SQL